### PR TITLE
[pthread-stubs] allow building release only

### PR DESCRIPTION
**Pull request by m-kuhn**
**Describe the pull request**

- #### What does your PR fix?
  Fixes building `pthread-stubs` release only

### CODE REVIEW

File: ports/pthread-stubs/portfile.cmake [hunk 1]
```diff
@@ -33,19 +33,21 @@ if(EXISTS "${CURRENT_INSTALLED_DIR}/lib/pthreadGC3.lib")
...
-string(REPLACE "Cflags: -pthread" "Cflags: " _contents "${_contents}")
-if(EXISTS "${CURRENT_INSTALLED_DIR}/debug/lib/pthreadVC3.lib")
-    string(REPLACE "Libs: -pthread" "Libs: -lpthreadVC3" _contents "${_contents}")
-endif()
-if(EXISTS "${CURRENT_INSTALLED_DIR}/debug/lib/pthreadGC3.lib")
-    string(REPLACE "Libs: -pthread" "Libs: -lpthreadGC3" _contents "${_contents}")
-endif()
-if(EXISTS "${CURRENT_INSTALLED_DIR}/debug/lib/pthreadVC3d.lib")
-    string(REPLACE "Libs: -pthread" "Libs: -lpthreadVC3d" _contents "${_contents}")
+if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
```
Comment (Neumann-A) [inline ports/pthread-stubs/portfile.cmake:36]:
```suggestion
if(NOT VCPKG_BUILD_TYPE)
```

File: ports/pthread-stubs/portfile.cmake [hunk 2]
```diff
@@ -33,19 +33,21 @@ if(EXISTS "${CURRENT_INSTALLED_DIR}/lib/pthreadGC3.lib")
...
+    endif()
+    if(EXISTS "${CURRENT_INSTALLED_DIR}/debug/lib/pthreadGC3.lib")
+        string(REPLACE "Libs: -pthread" "Libs: -lpthreadGC3" _contents "${_contents}")
+    endif()
+    if(EXISTS "${CURRENT_INSTALLED_DIR}/debug/lib/pthreadVC3d.lib")
+        string(REPLACE "Libs: -pthread" "Libs: -lpthreadVC3d" _contents "${_contents}")
+    endif()
+    if(EXISTS "${CURRENT_INSTALLED_DIR}/debug/lib/pthreadGC3d.lib")
+        string(REPLACE "Libs: -pthread" "Libs: -lpthreadGC3d" _contents "${_contents}")
+    endif()
```
Comment (MonicaLiu0311) [inline ports/pthread-stubs/portfile.cmake:51]: vcpkg_replace_string("${CURRENT_INSTALLED_DIR}/debug/lib/pthreadGC3d.lib" "Libs: -pthread" "Libs: -lpthreadGC3d")

I think using this function can make the file more concise, the specific function is defined as follows:
[docs/maintainers/vcpkg_replace_string.md](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_replace_string.md)
[scripts/cmake/vcpkg_replace_string.cmake](https://github.com/microsoft/vcpkg/blob/master/scripts/cmake/vcpkg_replace_string.cmake)

**Comment by MonicaLiu0311** [comment]
The file contains at least one deprecated function, we recommend the new function if you find it convenient:
`vcpkg_install_cmake` -> `vcpkg_cmake_install` (from port `vcpkg-cmake`)
`vcpkg_build_cmake` -> `vcpkg_cmake_build `(from port `vcpkg-cmake`)
`vcpkg_configure_cmake` -> `vcpkg_cmake_configure` (Please remove the option PREFER_NINJA) (from port `vcpkg-cmake`)
`vcpkg_fixup_cmake_targets` -> `vcpkg_cmake_config_fixup` (from port `vcpkg-cmake-config`)
`vcpkg_extract_source_archive_ex` -> [`vcpkg_extract_source_archive`](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_extract_source_archive.md)
`vcpkg_build_msbuild` -> [`vcpkg_install_msbuild`](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_msbuild.md)
`vcpkg_copy_tool_dependencies` -> [`vcpkg_copy_tools`](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_copy_tools.md)
`vcpkg_apply_patches` should be replaced by the `PATCHES` arguments to the "extract" helpers (e.g. [`vcpkg_from_github()`](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_from_github.md)

In the ports that use the new function, you have to add the corresponding dependencies:
```
{
  "name": "vcpkg-cmake",
  "host": true
},
{
  "name": "vcpkg-cmake-config",
  "host": true
}
```

**Comment by m-kuhn** [comment]
This PR solves one issue with this port. Further comments in here are possibly good but out of scope.

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**